File: codes/cpp/chap08/answer_A54.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

int Q;
int QueryType[100009], y[100009];
string x[100009];
map<string, int> Map;

int main() {
	// 入力
	cin >> Q;
	for (int i = 1; i <= Q; i++) {
		cin >> QueryType[i];
		if (QueryType[i] == 1) cin >> x[i] >> y[i];
		if (QueryType[i] == 2) cin >> x[i];
	}

	// クエリの処理
	for (int i = 1; i <= Q; i++) {
		if (QueryType[i] == 1) Map[x[i]] = y[i];
		if (QueryType[i] == 2) cout << Map[x[i]] << endl;
	}
	return 0;
}

File: codes/java/chap08/answer_A54.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 入力
		Scanner sc = new Scanner(System.in);
		int Q = sc.nextInt();
		int[] queryType = new int[Q + 1];
		String[] x = new String[Q + 1];
		int[] y = new int[Q + 1];
		for (int i = 1; i <= Q; i++) {
			queryType[i] = sc.nextInt();
			if (queryType[i] == 1) {
				x[i] = sc.next();
				y[i] = sc.nextInt();
			}
			if (queryType[i] == 2) {
				x[i] = sc.next();
			}
		}

		// クエリの処理
		Map<String, Integer> scoreMap = new HashMap<>();
		for (int i = 1; i <= Q; i++) {
			if (queryType[i] == 1) {
				scoreMap.put(x[i], y[i]);
			}
			if (queryType[i] == 2) {
				System.out.println(scoreMap.get(x[i]));
			}
		}
	}
}